rct229/rulesets/ashrae9012019/section1/section1rule2.py

```diff
@@ -0,0 +1,115 @@
+            ),
+            rmrs_used_optional=produce_ruleset_model_instance(
+                USER=True,
+                BASELINE_0=True,
+                BASELINE_90=True,
+                BASELINE_180=True,
+                BASELINE_270=True,
+                PROPOSED=True,
+            ),
+            id="1-2",
+            description="The performance of the proposed design is calculated in accordance with Standard 90.1-2019 Appendix G, where Performance Cost Index = Proposed building performance (PBP) /Baseline building performance (BBP), "
+            "where both the PBP and the BBP include all end-use load components associated with the building when calculating the Performance Cost Index (PCI).",
+            ruleset_section_title="Performance Calculations",
+            standard_section="Section G1.2.2",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+        )
+
+    def get_calc_vals(self, context, data=None):
+        rmd_u = context.USER
+        rmd_b0 = context.BASELINE_0
+        rmd_b90 = context.BASELINE_90
+        rmd_b180 = context.BASELINE_180
+        rmd_b270 = context.BASELINE_270
+        rmd_p = context.PROPOSED
+
+        pci_set = []
+        pbp_set = []
+        bbp_set = []
+        for rmd in (rmd_u, rmd_b0, rmd_b90, rmd_b180, rmd_b270, rmd_p):
+            if rmd is not None:
+                pci_set.append(find_one("$.output.performance_cost_index", rmd))
+                pbp_set.append(
+                    find_one(
+                        "$.output.total_proposed_building_energy_cost_including_renewable_energy",
+                        rmd,
+                    )
+                )
+                bbp_set.append(
+                    find_one("$.output.baseline_building_performance_energy_cost", rmd)
+                )
+
+        pci_set = filter_(pci_set, lambda x: x is not None)
+        pbp_set = filter_(pbp_set, lambda x: x is not None)
+        bbp_set = filter_(bbp_set, lambda x: x is not None)
+
+        assert_(
+            len(pci_set) >= 1, "At least one `performance_cost_index` value must exist."
+        )
+        assert_(
+            len(pbp_set) >= 1,
+            "At least one `total_proposed_building_energy_cost_including_renewable_energy` value must exist.",
+        )
+        assert_(
+            len(bbp_set) >= 1,
+            "At least one `baseline_building_performance_energy_cost` value must exist.",
+        )
+
+        return {
+            "pci_set": list(set(pci_set)),
+            "pbp_set": list(set(pbp_set)),
+            "bbp_set": list(set(bbp_set)),
+        }
+
+    def rule_check(self, context, calc_vals=None, data=None):
+        pci_set = calc_vals["pci_set"]
+        pbp_set = calc_vals["pbp_set"]
+        bbp_set = calc_vals["bbp_set"]
+
+        return len(pci_set) == len(pbp_set) == len(bbp_set) == 1 and std_equal(
+            pbp_set[0] / bbp_set[0], pci_set[0]
+        )  # TODO: what if  bbp_set[0] == 0?
```

Comment:
Good catch. My suggestion is to add bbp_set[0] != 0 in the return conditions and also add a fail message for this case to say "Ruleset expects `baseline_building_performance_energy_cost` to be greater than 0"

Reply:
Good point. Addressed